File: src/test/java/com/github/qky666/selenidepom/test/java/mtp/pom/plainjavapom/CookiesBannerWidget.java
package com.github.qky666.selenidepom.test.java.mtp.pom.plainjavapom;

import com.codeborne.selenide.Condition;
import com.codeborne.selenide.SelenideElement;
import com.github.qky666.selenidepom.pom.Required;
import com.github.qky666.selenidepom.pom.LoadableKt;
import com.github.qky666.selenidepom.pom.Widget;

import java.time.Duration;

import static com.codeborne.selenide.Condition.disappear;

public class CookiesBannerWidget extends Widget {
    @Required public final SelenideElement cookiesText;
    @Required public final SelenideElement accept;

    public CookiesBannerWidget(SelenideElement self) {
        super(self);
        cookiesText = $("div.cli-bar-message");
        accept = $("a#cookie_action_close_header");
    }

    @Override
    public void customShouldLoadRequired(Duration timeout, String pomVersion, String lang) throws Throwable {
        super.customShouldLoadRequired(timeout, pomVersion, lang);
        cookiesText.shouldHave(Condition.text("Utilizamos cookies propias y de terceros para fines analíticos y para mostrarte publicidad personalizada en base a un perfil elaborado a partir de tus hábitos de navegación (por ejemplo, páginas visitadas)"), timeout);
    }

    public void acceptCookies() {
        LoadableKt.shouldLoadRequired(this).accept.click();
        should(disappear);
    }
}

File: cookiecutter/{{cookiecutter.repo_name}}/src/test/kotlin/{{ cookiecutter.group }}/pom/common/CookiesBannerWidget.kt
package {{ cookiecutter.group }}.pom.common

import com.codeborne.selenide.Condition.disappear
import com.codeborne.selenide.Condition.text
import com.codeborne.selenide.SelenideElement
import com.github.qky666.selenidepom.pom.LangConditionedElement
import com.github.qky666.selenidepom.pom.Required
import com.github.qky666.selenidepom.pom.Widget
import com.github.qky666.selenidepom.pom.shouldLoadRequired

class CookiesBannerWidget(self: SelenideElement) : Widget(self) {

    @Required val cookiesText = LangConditionedElement(
        find("div.cli-bar-message"),
        mapOf("es" to text("Utilizamos cookies propias y de terceros para fines analíticos y para mostrarte publicidad personalizada en base a un perfil elaborado a partir de tus hábitos de navegación (por ejemplo, páginas visitadas)")),
        false
    )

    @Required val accept =
        LangConditionedElement(find("a#cookie_action_close_header"), mapOf("es" to text("Aceptar")), false)

    fun acceptCookies() {
        shouldLoadRequired().accept.click()
        should(disappear)
    }
}
